txpool/txpool_test.go

```diff
@@ -1054,119 +988,10 @@ func TestDrop(t *testing.T) {
-		pool.EnableDev()
-
-		// send 1st tx
-		go func() {
-			err := pool.addTx(local, newTx(addr1, 0, 1))
-			assert.NoError(t, err)
-		}()
-		go pool.handleEnqueueRequest(<-pool.enqueueReqCh)
-
-		// save the promotion for later
-		prom := <-pool.promoteReqCh
-
-		// send 2nd tx
-		go func() {
-			err := pool.addTx(local, newTx(addr1, 1, 1))
-			assert.NoError(t, err)
-		}()
-		pool.handleEnqueueRequest(<-pool.enqueueReqCh)
-
-		// promote
-		pool.handlePromoteRequest(prom)
-
-		assert.Equal(t, uint64(2), pool.gauge.read())
-		assert.Equal(t, uint64(2), pool.accounts.get(addr1).getNonce())
-		assert.Equal(t, uint64(2), pool.accounts.get(addr1).promoted.length())
-
-		pool.Prepare()
-
-		// process 1st tx
-		tx := pool.Peek()
-		pool.Pop(tx)
-
-		assert.Equal(t, uint64(1), pool.gauge.read())
-		assert.Equal(t, uint64(1), pool.accounts.get(addr1).promoted.length())
-
-		/* demote 2nd tx */
-		tx = pool.Peek()
-
-		// save the add request
-		// originating in Demote
-		var addReq enqueueRequest
-		go func() {
-			addReq = <-pool.enqueueReqCh
-		}()
-
-		pool.Demote(tx)
-
-		// handle add request
-		go pool.handleEnqueueRequest(addReq)
-		pool.handlePromoteRequest(<-pool.promoteReqCh)
-
-		assert.Equal(t, uint64(1), pool.gauge.read())
-		assert.Equal(t, uint64(0), pool.accounts.get(addr1).enqueued.length())
-		assert.Equal(t, uint64(1), pool.accounts.get(addr1).promoted.length())
-	})
-
-	t.Run("demote will enqueue", func(t *testing.T) {
-		pool, err := newTestPool(nil)
-		assert.NoError(t, err)
-		pool.SetSigner(&mockSigner{})
-		pool.EnableDev()
-
-		// send 2 txs and promote them
-		go func() {
-			err := pool.addTx(local, newTx(addr1, 0, 1))
-			assert.NoError(t, err)
-		}()
-		go pool.handleEnqueueRequest(<-pool.enqueueReqCh)
-		prom := <-pool.promoteReqCh
-		go func() {
-			err := pool.addTx(local, newTx(addr1, 1, 1))
-			assert.NoError(t, err)
-		}()
-		pool.handleEnqueueRequest(<-pool.enqueueReqCh)
-		pool.handlePromoteRequest(prom)
-
-		assert.Equal(t, uint64(2), pool.gauge.read())
-		assert.Equal(t, uint64(2), pool.accounts.get(addr1).getNonce())
-		assert.Equal(t, uint64(2), pool.accounts.get(addr1).promoted.length())
 
-		pool.Prepare()
-
-		// drop first tx
-		tx := pool.Peek()
-		pool.Drop(tx) // this will rollback nonce
-
-		assert.Equal(t, uint64(1), pool.gauge.read())
-		assert.Equal(t, uint64(0), pool.accounts.get(addr1).getNonce())
-		assert.Equal(t, uint64(1), pool.accounts.get(addr1).promoted.length())
-
-		// demote second tx
-		tx = pool.Peek()
-		// save the add request
-		// originating in Demote
-		var addReq enqueueRequest
-		go func() {
-			addReq = <-pool.enqueueReqCh
-		}()
-
-		pool.Demote(tx)
-
-		// enqueue demoted
-		pool.handleEnqueueRequest(addReq)
-
-		assert.Equal(t, uint64(1), pool.gauge.read())
-		assert.Equal(t, uint64(1), pool.accounts.get(addr1).enqueued.length())
-		assert.Equal(t, uint64(0), pool.accounts.get(addr1).promoted.length())
-	})
+func TestDemote(t *testing.T) {
+	// TODO dbrajovic
```

Comment:
Can we do this as a part of this PR?

Reply:
We're removing a lot of tests, but not adding any :)

Reply:
The reason a lot of existing tests were removed is because the entire flow for demoted transactions is removed. Essentially, there is no need to do any handling (using Demote) for recoverable transactions:

https://github.com/0xPolygon/polygon-edge/blob/5bb6451d254dc465694f85543bc7cbaedadb3419/consensus/ibft/ibft.go#L699-L701

Since, previously, these transactions would just end up back in promote anyhow, now we're just shortcutting that by leaving the account as it is (and excluding it from further processing).

I've left TestDemote as a todo for when we eventually implement the pruning mechanism.